### PR TITLE
[SPARK-28543][DOCS][WebUI] Document Spark Jobs page

### PR DESCRIPTION
**Pull request by planga82**
## What changes were proposed in this pull request?

New documentation to explain in detail Web UI Jobs page and link it to monitoring page. New images are included to better explanation

image
image

## How was this patch tested?

This pull request contains only documentation. I have generated it using "jekyll build" to ensure that it's ok

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
ok to test

**Comment by planga82** [comment]
image
image

**Comment by dongjoon-hyun** [comment]
Thank you for your contribution, @planga82 .

File: docs/web-ui.md [hunk 1]
```diff
@@ -35,6 +35,64 @@ progress of all jobs and the overall event timeline. When you click on a job on
 page, you see the details page for that job. The details page further shows the event timeline,
 DAG visualization, and all stages of the job.
 
+The information that is displayed in this section is
+* User: Current spark user
```
Comment (srowen) [inline docs/web-ui.md]: spark -> Spark (and below)

File: docs/web-ui.md [hunk 2]
```diff
@@ -35,6 +35,64 @@ progress of all jobs and the overall event timeline. When you click on a job on
...
+* Total uptime: Time since spark application starts
```
Comment (srowen) [inline docs/web-ui.md]: starts -> started

File: docs/web-ui.md [hunk 3]
```diff
@@ -35,6 +35,64 @@ progress of all jobs and the overall event timeline. When you click on a job on
...
+* Scheduling mode: (More information in [job scheduling](job-scheduling.html#configuring-pool-properties) page)
```
Comment (srowen) [inline docs/web-ui.md]: Remove parentheses, or just say "See [job scheduling]"

File: docs/web-ui.md [hunk 4]
```diff
@@ -35,6 +35,64 @@ progress of all jobs and the overall event timeline. When you click on a job on
...
+  <img src="img/AllJobsPageDetail1.png" title="Basic info" alt="Basic info" width="20%"/>
+</p>
+
+* Event timeline: Displays in chronological order the events related to the executors (added, removed) and the jobs
+
+<p style="text-align: center;">
+  <img src="img/AllJobsPageDetail2.png" title="Event timeline" alt="Event timeline"/>
+</p>
+
+* Details of jobs grouped by status: Displays detailed information of the jobs including Job id, description (with a link to detailed job page), submitted time, duration, stages summary and tasks progress bar
```
Comment (srowen) [inline docs/web-ui.md]: id -> ID

File: docs/web-ui.md [hunk 5]
```diff
@@ -35,6 +35,64 @@ progress of all jobs and the overall event timeline. When you click on a job on
...
+<p style="text-align: center;">
+  <img src="img/AllJobsPageDetail3.png" title="Details of jobs grouped by status" alt="Details of jobs grouped by status"/>
+</p>
+
+
+When you click on a specific job, you can see the detailed information of this job.
+
+### Jobs detail
+
+This page displays the details of a specific job identified by its job id. The information we could find is:
```
Comment (srowen) [inline docs/web-ui.md]: Remove the "The information we could find is:"

File: docs/web-ui.md [hunk 6]
```diff
@@ -35,6 +35,64 @@ progress of all jobs and the overall event timeline. When you click on a job on
...
+* Job Status: (running, succeeded, failed)
+* Number of stages per status (active, pending, completed, skipped, failed)
+* Associated SQL Query: Link to the sql tab for this job
+* Event timeline: Displays in chronological order the events related to the executors (added, removed) and the stages of the job
+
+<p style="text-align: center;">
+  <img src="img/JobPageDetail1.png" title="Event timeline" alt="Event timeline"/>
+</p>
+
+* DAG visualization: Visual representation of the directed acyclic graph of this job where vertices represent the RDDs and the edges represent an operation to be applied on RDD.
```
Comment (srowen) [inline docs/web-ui.md]: RDDs or DataFrames

File: docs/web-ui.md [hunk 7]
```diff
@@ -35,6 +35,64 @@ progress of all jobs and the overall event timeline. When you click on a job on
...
+
+<p style="text-align: center;">
+  <img src="img/JobPageDetail2.png" title="DAG" alt="DAG" width="40%">
+</p>
+
+* List of stages (grouped by state active, pending, completed, skipped, and failed)
+	* Stage Id
```
Comment (srowen) [inline docs/web-ui.md]: Id -> ID

**Review by srowen — COMMENTED** [review]
I'm not against this, but it's mostly repeating what's written on the UI page and in tooltips. How much does this add?

**Comment by planga82** [comment]
Thanks for the review @srowen , I have fixed all comments
About your first question, sure, it doesn't add too much. When I saw the Jira issue I thought about how to drive it. I think it's a good idea to have documentation  about the web UI but I don't know what kind of documentation. Do you think it's better to add some examples? or explain what is a Job... but I think it's not the best place to explain what is a Job. Any idea to improve it?

**Comment by srowen** [comment]
Yeah I don't know right now. Maybe OK to just get a basic page down in the docs.

**Comment by zhengruifeng** [comment]
@srowen  I had some offline discussion with @gatorsmile ,  and the expected doc seems to be something like 
1, https://www.cloudera.com/documentation/enterprise/5-9-x/topics/operation_spark_applications.html#spark_monitoring
2, https://www.ibm.com/support/knowledgecenter/en/SS3H8V_1.1.0/com.ibm.izoda.v1r1.azka100/topics/azkic_c_webUIs.htm

the first step in my opinion is to add some basic explanation, example and figure for each tab, and we can refine it in the future.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**